From 1931eaf90e3f319698a20e895b6b570c11064449 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 28 Feb 2023 13:38:58 +0200
Subject: [PATCH] [SW-110215] pcie_pif: skip wait for d1 when rst disabled

During a normal FLR sequence D1 is being reset and PIF ARC is waiting
for it to become ready by polling on psoc2pif_die_ready register.
This register is updated by PSOC only when D1 status changes to ready,
and it is zeroized by PIF ARC when FLR flow ends.
In case FLR reset is disabled, D1 status doesn't change because it is
not being reset, and psoc2pif_die_ready would remain zeroized. In that
case PIF should not poll on this register.

Change-Id: I40dad082b22749b884aa055b2f2a0ca3f3259882
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/pcie_pif/pcie_pif.c b/app/pcie_pif/pcie_pif.c
index 08e0f1e..f06426a 100644
--- a/app/pcie_pif/pcie_pif.c
+++ b/app/pcie_pif/pcie_pif.c
@@ -98,6 +98,9 @@
 
 static void wait_for_secondary_die(void)
 {
+	if (!pif_flr_rst_en())
+		return;
+
 	while (!regmap_readl(pif_arc.psoc2pif_die_ready)) {
 		/* meassuring timeout from FLR event start */
 		if (pif_ts_timeout_expired(0, FLR_SAFE_TIMEOUT_US)) {
